### PR TITLE
Fix wrong partitionsSpec type names in the document

### PR DESCRIPTION
**Pull request by jihoonson**
### Description

The doc for partitionsSpecs was changed in https://github.com/apache/incubator-druid/pull/8141, but it's using wrong type names for them. This PR fixes those wrong names.

Also, this PR adds some deprecated JSON properties in indexTuningConfig for backward compatibility.

<hr>

This PR has:
- [x] been self-reviewed.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added unit tests or modified existing tests to cover new code paths.

### CODE REVIEW

File: docs/content/ingestion/native_tasks.md [hunk 1]
```diff
@@ -195,9 +195,12 @@ The tuningConfig is optional and default parameters will be used if no tuningCon
...
 |maxBytesInMemory|Used in determining when intermediate persists to disk should occur. Normally this is computed internally and user does not need to set it. This value represents number of bytes to aggregate in heap memory before persisting. This is based on a rough estimate of memory usage and not actual usage. The maximum heap memory usage for indexing is maxBytesInMemory * (2 + maxPendingPersists)|1/6 of max JVM memory|no|
-|partitionsSpec|Defines how to partition the segments in a timeChunk, see [PartitionsSpec](#partitionsspec)|`dynamic` if `forceGuaranteedRollup` = false, `hashed` if `forceGuaranteedRollup` = true|no|
+|maxTotalRows|Deprecated. Use `partitionsSpec` instead. Total number of rows in segments waiting for being pushed. Used in determining when intermediate pushing should occur.|20000000|no|
+|numShards|Deprecated. Use `partitionsSpec` instead. Directly specify the number of shards to create. If this is specified and 'intervals' is specified in the granularitySpec, the index task can skip the determine intervals/partitions pass through the data. numShards cannot be specified if maxRowsPerSegment is set.|null|no|
```
Comment (clintropolis) [inline docs/content/ingestion/native_tasks.md]: nit: maybe should be "`" around the other settings, `granularitySpec `, `numShards`, `maxRowsPerSegment`

File: docs/content/ingestion/native_tasks.md [hunk 2]
```diff
@@ -599,9 +602,13 @@ The tuningConfig is optional and default parameters will be used if no tuningCon
...
```
Comment (clintropolis) [inline docs/content/ingestion/native_tasks.md]: same about "`" and references to other settings

File: docs/content/ingestion/native_tasks.md (hunk 1 above)
Comment (jihoonson) [reply docs/content/ingestion/native_tasks.md]: Thanks, fixed.

File: docs/content/ingestion/native_tasks.md (hunk 2 above)
Comment (jihoonson) [reply docs/content/ingestion/native_tasks.md]: Fixed.

**Comment by jihoonson** [comment]
@clintropolis thanks for taking a look. Fixed conflicts.

**Pull request merged**